File: src/main/java/me/sd_master92/customvoting/listeners/VotifierListener.java
package me.sd_master92.customvoting.listeners;

import com.vexsoftware.votifier.model.VotifierEvent;
import me.sd_master92.customvoting.Main;
import me.sd_master92.customvoting.subjects.CustomVote;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;

public class VotifierListener implements Listener
{
    private final Main plugin;

    public VotifierListener(Main plugin)
    {
        this.plugin = plugin;
    }

    @EventHandler(priority = EventPriority.NORMAL)
    public void onVotifierEvent(VotifierEvent event)
    {
        new CustomVote(plugin, event.getVote());
    }
}

File: src/main/kotlin/me/sd_master92/customvoting/listeners/VotifierListener.kt
package me.sd_master92.customvoting.listeners

import com.vexsoftware.votifier.model.VotifierEvent
import me.sd_master92.customvoting.CV
import me.sd_master92.customvoting.constants.models.VoteSiteUUID
import me.sd_master92.customvoting.subjects.CustomVote
import me.sd_master92.customvoting.subjects.VoteSite
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener

class VotifierListener(private val plugin: CV) : Listener
{
    @EventHandler(priority = EventPriority.NORMAL)
    fun onVotifierEvent(event: VotifierEvent)
    {
        val vote = event.vote
        CustomVote(plugin, vote)

        VoteSite.register(plugin, VoteSiteUUID(vote.serviceName))
    }
}